internal/daemonset/daemonset.go

```diff
@@ -421,7 +419,7 @@ func MakeUnloadCommand(spec kmmv1beta1.ModprobeSpec, modName string) []string {
 
 	fwUnloadCommand := ""
 	if fw := spec.FirmwarePath; fw != "" {
-		fwUnloadCommand = fmt.Sprintf(" && rm -rf %s/%s", nodeVarLibFirmwarePath, modName)
+		fwUnloadCommand = fmt.Sprintf(" && cd %s && find |sort -r |xargs -I{} rm -d %s/$(basename %s)/{}", fw, nodeVarLibFirmwarePath, fw)
```

Comment:
why cannot we just remove the file with rm -f <fileName> command? We have the directory (nodeVarLibFirmwarePath) and the file name/subpath (spec.FirmwarePath)...

Reply:
If we try to remove the whole directory directly, it could be that we remove files that were not copied for the Module being unloaded. The latter would lead to another Module being broken. For example:

- `Module A` includes in its module-loader image this `/opt/lib/firmware/vendor/firmware_a.bin` fw and the Module's `FirmwarePath: /opt/lib/firmware/vendor` 
- `Module B`  includes in its module-loader image this `/opt/lib/firmware/vendor/firmware_b.bin` fw and the Module's `FirmwarePath: /opt/lib/firmware/vendor`
- After loading both Modules, the host firmware directory would be `/var/lib/firmware/vendor/{firmware_a.bin,firmware_b.bin}`
- By unloading `Module A` and `rm`ing the `/var/lib/firmware/vendor/` dir, we would also remove Module B's firmware

What do you think? Does this make sense?